federationsender/queue/queue.go

```diff
@@ -136,7 +136,7 @@ func (oqs *OutgoingQueues) SendEvent(
 	}
 
 	for _, destination := range destinations {
-		oqs.getQueue(destination).sendEvent(nid)
+		go oqs.getQueue(destination).sendEvent(nid)
```

Comment:
Why would `sendEvent` block in the new implementation? This could spawn a lot of goroutines, and even though Go supports millions of them this is not a good way to go about it I think.

Reply:
You're right - with the `select` it shouldn't block so I've taken the `go` out.